src/proc_cpuview.c

```diff
@@ -920,6 +947,8 @@ int proc_cpuinfo_read(char *buf, size_t size, off_t offset,
 	bool use_view;
 	char *cache = d->buf;
 	size_t cache_size = d->buflen;
+	__u32 host_personality, caller_personality;
+	call_cleaner(restore_personality) bool personality_changed = false;
```

Comment:
Ah, I missed that. I think wrapping `proc_cpuinfo_read()` might be cleaner though.

Reply:
I've introduced the wrapper function `proc_read_with_personality` for such cases.